plugins/in_exec/in_exec.c

```diff
@@ -69,10 +69,12 @@ static int in_exec_collect(struct flb_input_instance *ins,
     if (ctx->parser) {
         while (fgets(ctx->buf, ctx->buf_size, cmdp) != NULL) {
             str_len = strnlen(ctx->buf, ctx->buf_size);
-            ctx->buf[str_len - 1] = '\0'; /* chomp */
+            if (ctx->buf[str_len - 1] == '\n') {
+                ctx->buf[str_len - 1] = '\0'; /* chomp */
```

Comment:
I think you might want to decrement str_len here since you're effectively shortening the string by one so maybe this would be a good alternative since it decrement before using `str_len` which would also change it for line 77 (same applies to the code in line 108)

```
ctx->buf[--str_len] = '\0'; /* chomp */
```

Reply:
Yes, will make the code more readable. Will change.